connection/test-utils/src/blockchain.rs

```diff
@@ -33,15 +33,24 @@ pub struct MockBlockchainConnection<L: Ledger + Sync> {
 
     /// Proposed transactions.
     pub proposed_txs: Vec<Tx>,
+
+    /// The fee map for this peer.
+    pub fee_map: FeeMap,
 }
 
 impl<L: Ledger + Sync> MockBlockchainConnection<L> {
-    pub fn new(uri: ConsensusClientUri, ledger: L, latency_millis: u64) -> Self {
+    pub fn new(
+        uri: ConsensusClientUri,
+        ledger: L,
+        latency_millis: u64,
+        fee_map: Option<FeeMap>,
+    ) -> Self {
         Self {
             uri,
             ledger,
             latency_millis,
             proposed_txs: Vec::new(),
+            fee_map: fee_map.unwrap_or_default(),
```

Comment:
Since `FeeMap` implements `Default`, it seems fine to require it as a non-optional argument, and callers can pass `FeeMap::default()`
```suggestion
        fee_map: FeeMap,
    ) -> Self {
        Self {
            uri,
            ledger,
            latency_millis,
            proposed_txs: Vec::new(),
            fee_map,
```

Reply:
Good point! thanks :)